src/components/AddPaymentMethodMenu.js

```diff
@@ -9,54 +9,166 @@ import styles from '../styles/styles';
+import ROUTES from '../ROUTES';
+import Navigation from '../libs/Navigation/Navigation';
 
-const propTypes = {
-    isVisible: PropTypes.bool.isRequired,
-    onClose: PropTypes.func.isRequired,
-    anchorPosition: PropTypes.shape({
-        top: PropTypes.number,
-        left: PropTypes.number,
-    }),
 
+const propTypes = {
     /** Username for PayPal.Me */
     payPalMeUsername: PropTypes.string,
 
+    /** Type to filter the payment Method list */
+    filterType: PropTypes.oneOf([CONST.PAYMENT_METHODS.DEBIT_CARD, CONST.PAYMENT_METHODS.BANK_ACCOUNT, '']),
+
+    /** Are we loading payments from the server? */
+    isLoadingPayments: PropTypes.bool,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
-    anchorPosition: {},
     payPalMeUsername: '',
+    filterType: '',
+    isLoadingPayments: false,
 };
 
-const AddPaymentMethodMenu = props => (
-    <Popover
-        isVisible={props.isVisible}
-        onClose={props.onClose}
-        anchorPosition={props.anchorPosition}
-    >
-        <MenuItem
-            title={props.translate('common.bankAccount')}
-            icon={Expensicons.Bank}
-            onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.BANK_ACCOUNT)}
-            wrapperStyle={styles.pr15}
-        />
-        <MenuItem
-            title={props.translate('common.debitCard')}
-            icon={Expensicons.CreditCard}
-            onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.DEBIT_CARD)}
-            wrapperStyle={styles.pr15}
-        />
-        {!props.payPalMeUsername && (
-            <MenuItem
-                title={props.translate('common.payPalMe')}
-                icon={Expensicons.PayPal}
-                onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.PAYPAL)}
-                wrapperStyle={styles.pr15}
-            />
-        )}
-    </Popover>
-);
+class AddPaymentMethodMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isAddPaymentMenuActive: false,
+            anchorPosition: {
+                top: 0,
+                left: 0,
+            },
+        };
+        this.addPaymentMethodPressed = this.addPaymentMethodPressed.bind(this);
+        this.addPaymentMethodTypePressed = this.addPaymentMethodTypePressed.bind(this);
+        this.hideAddPaymentMenu = this.hideAddPaymentMenu.bind(this);
+    }
+
+    /**
+     * Get the AddPaymentMethod Button title
+     * @returns {String}
+     */
+    getAddPaymentMethodButtonTitle() {
+        switch (this.props.filterType) {
+            case CONST.PAYMENT_METHODS.BANK_ACCOUNT:
+                return this.props.translate('paymentMethodList.addBankAccount');
+            case CONST.PAYMENT_METHODS.DEBIT_CARD:
+                return this.props.translate('paymentMethodList.addDebitCard');
+            default: break;
+        }
+        return this.props.translate('paymentMethodList.addPaymentMethod');
+    }
```

Comment:
I'd suggest calling it `AddPaymentMethodButton`

Reply:
> AddPaymentMethodMenu itself shouldn't be modified because we need the popover to open when the "Transfer Balance" button is eventually pressed. 

I didn't get it and I don't have access to that issue. But I got the idea.

Reply:
Ah I'm sorry, I forgot that you cannot see those issues.